internal/pkg/kube/reconciler_manager.go

```diff
@@ -42,12 +43,15 @@ func NewReconcilerManager(pushRequests <-chan xds.PushRequest, reconcilers ...Re
 }
 
 func (rm *ReconcilerManager) ReconcileAll(ctx context.Context) error {
+	var reconcileErrs error
+
 	for _, r := range rm.reconcilers {
 		if err := r.Reconcile(ctx); err != nil {
-			return err
+			reconcileErrs = errors.Join(err)
```

Comment:
If there are multiple errors should they all be joined?
```suggestion
			reconcileErrs = errors.Join(err, reconcileErrs)
```